modules/on_gke/main.tf

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_project_service" "gcr" {
+  project            = "${var.project_id}"
+  service            = "containerregistry.googleapis.com"
+  disable_on_destroy = "false"
+}
+
+//*****************************************
+//  Create Keys for Forseti server and
+//  client service accounts.
+//*****************************************
+
+resource "google_service_account_key" "server_key" {
+  service_account_id = "${var.forseti_server_service_account}"
+}
+
+resource "google_service_account_key" "client_key" {
+  service_account_id = "${var.forseti_client_service_account}"
+}
+
+//*****************************************
+//  Create Keys for Forseti server and
+//  client service accounts.
+//*****************************************
+
+resource "google_project_iam_member" "cluster_service_account-storage_reader" {
+  count   = "${var.gke_service_account == "create" ? 1 : 0}"
+  project = "${var.project_id}"
+  role    = "roles/storage.objectViewer"
+  member  = "serviceAccount:${var.gke_service_account_name}"
+}
+
+// get forseti server config contents
+data "google_storage_object_signed_url" "file_url" {
+  bucket = "${var.forseti_server_bucket}"
+  path = "configs/forseti_conf_server.yaml"
+}
+
+data "http" "server_config_contents" {
+  url        = "${data.google_storage_object_signed_url.file_url.signed_url}"
+  depends_on = ["data.google_storage_object_signed_url.file_url"]
+}
+
+//*****************************************
+//  Setup the Kubernetes Provider
+//*****************************************
+
+data "google_client_config" "default" {}
+
+provider "kubernetes" {
```

Comment:
HashiCorp recommends that [provider configuration be handled by the calling module](https://www.terraform.io/docs/configuration-0-11/modules.html#providers-within-modules), just like how the Google providers are configured in the example modules. It looks like this should be possible with both of the providers currently in this submodule as they don't depend on any contextual inputs.

Reply:
Okeedokee, I think I understand what you are saying.